PhoneBook.Web/Classes/PhoneNumberCell.cs

```diff
@@ -11,46 +12,57 @@
 
 namespace PhoneBook.Web.Classes
 {
-
   public class PhoneNumberCell : BocCustomColumnDefinitionCell
   {
     public int MaxPhoneNumbers { get; set; }
     public bool Commit { get; set; }
 
     protected override void Render (HtmlTextWriter writer, BocCustomCellRenderArguments arguments)
     {
-      var propertyPath = arguments.ColumnDefinition.GetPropertyPath();
-      var bo = arguments.BusinessObject;
+      var renderedPhoneNumbers = GetPhoneNumbers(arguments)
+        .Take (MaxPhoneNumbers)
+        .Select (phoneNumber => $"<a href=\"#\" onclick=\"{GetPostBackClientEvent (phoneNumber.ID.ToString())}\">{HttpUtility.HtmlEncode (phoneNumber.DisplayName)}</a>");
 
-      var phoneNumbers = (IList) propertyPath.GetValue (bo, false, true);
-      for (int i = 0; i < MaxPhoneNumbers && i < phoneNumbers.Count; ++i)
+      foreach (var renderedPhoneNumber in renderedPhoneNumbers)
       {
-        var phoneNumber = (BindableDomainObject) phoneNumbers[i];
-        string renderedLink = String.Format ("<a href=\"#\" onclick=\"{0}\">{1}</a>",
-                                             GetPostBackClientEvent (phoneNumber.ID.ToString()),
-                                             HttpUtility.HtmlEncode (phoneNumber.DisplayName));
-        writer.Write (renderedLink);
-        writer.Write ("<br>");
+        writer.Write (renderedPhoneNumber);
+        writer.Write ("<br/>");
       }
     }
 
+    private static IEnumerable<BindableDomainObject> GetPhoneNumbers (BocCustomCellRenderArguments arguments)
+    {
+      var propertyPath = arguments.ColumnDefinition.GetPropertyPath();
+      var businessObject = arguments.BusinessObject;
+      var phoneNumbers = propertyPath.GetValue (businessObject, false, true) as IEnumerable ?? new BindableDomainObject[] { };
```

Comment:
This bit here is hardcoded to expect Phonenumbers. A hard caase to at least IEnumerable<BindableDomainObject> is okay.